pkg/activator/handler/transport.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handler
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+
+	"golang.org/x/net/http2"
+	"knative.dev/control-protocol/pkg/certificates"
+	netcfg "knative.dev/networking/pkg/config"
+	pkgnet "knative.dev/pkg/network"
+	activatorconfig "knative.dev/serving/pkg/activator/config"
+)
+
+func verifySanConnection(san string) func(tls.ConnectionState) error {
+	return func(cs tls.ConnectionState) error {
+		if cs.PeerCertificates == nil {
+			return errors.New("server certificate not verified during VerifyConnection")
```

Comment:
`Server certificate could not be verified during VerifyConnection. No PeerCertificates provided`